src/Microsoft.ML/Runtime/EntryPoints/MacroUtils.cs

```diff
@@ -186,13 +186,29 @@ public static TrainerKinds SignatureTypeToTrainerKind(Type sigType)
         public static TrainerKinds[] SignatureTypesToTrainerKinds(IEnumerable<Type> sigTypes) =>
             sigTypes.Select(SignatureTypeToTrainerKind).ToArray();
 
-        public static string GetTrainerName(TrainerKinds kind) => TrainerKindDict[kind].TrainerFunctionName;
+        private static string GetTrainerName(TrainerKinds kind) => TrainerKindDict[kind].TrainerFunctionName;
 
         public static T TrainerKindApiValue<T>(TrainerKinds trainerKind)
         {
             if (Enum.GetName(typeof(TrainerKinds), trainerKind) is string name)
                 return (T)Enum.Parse(typeof(T), name);
             throw new Exception($"Could not interpret enum value: {trainerKind}");
         }
+
+        public static bool IsTrainerOfKind(Type type, TrainerKinds trainerKind)
+        {
+            if (type.Name == "BinaryLogisticRegressor")
```

Comment:
>type.Name == "BinaryLogisticRegressor" [](start = 16, length = 38)

All of these string operations make me nervous.

Is it at all reasonable to change this (and other) type operations to just be `type == typeof(BinaryLogisticRegressor)` or at least rely on some other reference?

Reply:
Is there also any way we can write a test for this, so we can detect when something like this happens again?

---
In reply to: [187422829](https://github.com/dotnet/machinelearning/pull/113#discussion_r187422829) [](ancestors = 187422829)